ImageAnalysis/image_analysis/analyzers/UC_GenericLightSpecCam.py

```diff
@@ -67,10 +82,30 @@ def __init__(self,
         self.optimization_central_wavelength = float(optimization_central_wavelength)
         self.optimization_bandwidth_wavelength = float(optimization_bandwidth_wavelength)
 
+        self.pointing_correction_bool = bool(pointing_correction_bool)
+
+        # if spectral correction information available, do now to avoid redundant calculations
+        # if not available, retrieval and calculation performed in analyze_image
+        self.spectral_correction_bool = bool(spectral_correction_bool)
+        if self.spectral_correction_bool and spectral_correction_files is not None:
+            self.spectral_correction_files = list(spectral_correction_files)
+            self.spectral_correction_data = self.get_spectral_correction_data()
+        else:
+            self.spectral_correction_files = list()
+            self.spectral_correction_data = None
```

Comment:
I think like we talked about, if `self.spectral_correction_data` is only calculated in the `__init__` function, then this will not get updated when the config files are loaded using the `configure()` method.

I would suggest moving this logic somehow to the `analyze_image()` function.

Reply:
In that case, consider specifying a configure() method in this class that runs the parent class's configure() plus get_spectral_correction_data() : 
```python
def configure(**kwargs):
    super().configure(**kwargs)
    self.spectral_correction_data = self.get_spectral_correction_data()
```

Again, we want to avoid doing any "one-time" configuration in analyze_image(), since the same instance of this class will be used to call analyze_image() many times.

Reply:
I like this solution!